pkg/controller/node/nodecontroller.go

```diff
@@ -514,14 +514,20 @@ func (nc *NodeController) reconcileNodeCIDRs(nodes *api.NodeList) {
 	}
 	for _, node := range nodes.Items {
 		if node.Spec.PodCIDR == "" {
+			// Re-GET node (because ours might be stale by now).
+			n, err := nc.kubeClient.Core().Nodes().Get(node.Name)
+			if err != nil {
+				glog.Errorf("Failed to get node %q: %v", node.Name, err)
+				continue
+			}
 			podCIDR, found := availableCIDRs.PopAny()
 			if !found {
-				nc.recordNodeStatusChange(&node, "CIDRNotAvailable")
+				nc.recordNodeStatusChange(n, "CIDRNotAvailable")
 				continue
 			}
-			glog.V(4).Infof("Assigning node %s CIDR %s", node.Name, podCIDR)
-			node.Spec.PodCIDR = podCIDR
-			if _, err := nc.kubeClient.Core().Nodes().Update(&node); err != nil {
+			glog.V(4).Infof("Assigning node %s CIDR %s", n.Name, podCIDR)
+			n.Spec.PodCIDR = podCIDR
+			if _, err := nc.kubeClient.Core().Nodes().Update(n); err != nil {
```

Comment:
That feels scary. If there was a meaningful update in between my read and write, wouldn't I clobber it?